src/main/java/nextstep/subway/domain/SectionLineUp.java

```diff
@@ -0,0 +1,155 @@
+package nextstep.subway.domain;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Embeddable;
+import javax.persistence.FetchType;
+import javax.persistence.OneToMany;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * 노선 구간 목록을 관리하는 일급 컬렉션
+ */
+@Embeddable
+public class SectionLineUp {
+
+    private static final int START_INDEX = 0;
+
+    @OneToMany(mappedBy = "line", orphanRemoval = true, cascade = CascadeType.ALL, fetch = FetchType.LAZY)
+    private List<Section> sectionList = new ArrayList<>();
+
+    protected SectionLineUp() {
+    }
+
+    public List<Section> getSectionList(Comparator<Section> comparator) {
+        sectionList.sort(comparator);
+        return sectionList;
+    }
+
+    public void addSection(Section section) {
+        if (sectionList.isEmpty()) {
+            section.updateOrder(new Order(START_INDEX));
+            sectionList.add(section);
+            return;
+        }
+        validUnknownStation(section);
+        validSameSection(section);
+        add(section);
+    }
+
+    private void add(Section section) {
+        boolean isEndUpStation = isEndUpStation(section);
+        boolean isEndDownStation = isEndDownStation(section);
+        if (isEndUpStation) {
+            addEndUpStation(section);
+        }
+        if (isEndDownStation) {
+            addEndDownStation(section);
+        }
+        if (!isEndUpStation && !isEndDownStation) {
+            createInternal(section);
+        }
```

Comment:
```suggestion
        if (isEndUpStation) {
            addEndUpStation(section);
            return;
        }
        if (isEndDownStation) {
            addEndDownStation(section);
            return;
        }
        createInternal(section);
```
early return 하는건 어떨까요? 😀

Reply:
현재는 `상행 종점`에 등록되는 경우, `하행 종점`에 등록되는 경우, `구간 내부`에 등록되는 경우 `세 가지 케이스 중 하나의 케이스`로 분류 되는데, 케이스 여부에 상관 없이 모든 케이스 검증을 수행하고 있어, 불필요한 리소스를 사용하고 있었군요 !
피드백 반영하였습니다
감사합니다 ! 🙇🏻‍